gtsam/base/Vector.h

```diff
@@ -75,6 +76,17 @@ static_assert(
   "Error: GTSAM was built against a different version of Eigen");
 #endif
 
+/**
+ * Numerically stable function for comparing if floating point values are equal
+ * within epsilon tolerance.
```

Comment:
I think it is good to add comments on the behavior when we encounter NaNs and infinitys.

Reply:
agreed. Just recap very succinctly.

Reply:
Good point. I'll add it now.